sympy/physics/vector/frame.py

```diff
@@ -884,10 +884,10 @@ def orient_body_fixed(self, parent, angles, rotation_order):
             templist = kinematic_equations([u1, u2, u3], [q1, q2, q3],
                                            'body', rot_order)
             templist = [expand(i) for i in templist]
-            td = solve(templist, [u1, u2, u3])
-            u1 = expand(td[u1])
-            u2 = expand(td[u2])
-            u3 = expand(td[u3])
+            td = solve(templist, [u1, u2, u3], dict = True)
+            u1 = expand(td[0][u1])
+            u2 = expand(td[0][u2])
+            u3 = expand(td[0][u3])
```

Comment:
I have fixed the error which symengine was giving earlier. Although, it looks like it is unable to find a solution to the equations it is trying to solve which returns empty list causing index error.

Reply:
@sidhu1012 I have been unable to find out why the `td[0]` is giving a keyerror. Maybe someone familiar with the code can help me there. But I was successful in recreating the dictionary functionality using sets. I did the following changes and ran symengine test and it has passed.
```python
            td = solve(templist, [u1, u2, u3], set = True)
            u1 = expand(list(td[1])[0][td[0].index(u1)])
            u2 = expand(list(td[1])[0][td[0].index(u2)])
            u3 = expand(list(td[1])[0][td[0].index(u3)])
```

Reply:
> I have found that changing u1, u2, u3 to dynamicsymbols does yield a result but idk why then it cannot find u1 as a dictionary index.

Can you show the code needed to setup the inputs to solve that leads to this problem?

There is probably a bug but it needs a reproducer (that doesn't involve calling the functions in the mechanics module).

Reply:
It is indeed a bug and a very odd one at that. Somehow changing some other function changed whether it passed the test or not. I can open a new issue describing all the oddities that I experienced with it.